src/components/core/geoChart/GeoChartLegendRenderer.tsx

```diff
@@ -4,66 +4,190 @@ import Measure, { ContentRect, MeasuredComponentProps } from "react-measure";
-import { BOTTOM } from "../../visualizations/chart/legend/PositionTypes";
-import { IGeoData, IPushpinCategoryLegendItem } from "../../../interfaces/GeoChart";
+import Paging from "../../visualizations/chart/legend/Paging";
+import { BOTTOM, LEFT, RIGHT } from "../../visualizations/chart/legend/PositionTypes";
+import { PositionType } from "../../visualizations/typings/legend";
+import { IAvailableLegends, IGeoData, IPushpinCategoryLegendItem } from "../../../interfaces/GeoChart";
 import { isFluidLegendEnabled } from "../../../helpers/geoChart/common";
 import { getAvailableLegends } from "../../../helpers/geoChart/data";
 
+const HEIGHT_OF_COLOR_LEGEND = 210;
+
 export interface IGeoChartLegendRendererProps {
     categoryItems?: IPushpinCategoryLegendItem[]; // used for Category legend
     format?: string;
     geoData?: IGeoData; // used for Color/Size legend
     height?: number;
     locale?: string;
     colorLegendValue: string;
-    position?: string;
+    position?: PositionType;
     responsive?: boolean;
     showFluidLegend?: boolean;
     onItemClick?: (item: IPushpinCategoryLegendItem) => void;
 }
 
-export default function GeoChartLegendRenderer(props: IGeoChartLegendRendererProps): JSX.Element {
-    const { categoryItems, geoData, position, responsive, showFluidLegend } = props;
-    const { hasCategoryLegend, hasColorLegend, hasSizeLegend } = getAvailableLegends(categoryItems, geoData);
-
-    const isLegendVisible = hasCategoryLegend || hasColorLegend || hasSizeLegend;
-    if (!isLegendVisible) {
-        return null;
-    }
+function getClassnames(props: IGeoChartLegendRendererProps, availableLegends: IAvailableLegends): string {
+    const { position, responsive, showFluidLegend } = props;
+    const { hasSizeLegend } = availableLegends;
 
     const isFluidLegend = isFluidLegendEnabled(responsive, showFluidLegend);
     const isBottomPosition = isFluidLegend || position === BOTTOM;
 
-    const classNames = cx("geo-legend", "s-geo-legend", {
+    return cx("geo-legend", "s-geo-legend", "viz-legend", {
         "viz-fluid-legend-wrap": isFluidLegend,
         "viz-static-legend-wrap": !isFluidLegend,
+        static: !isFluidLegend,
         "has-size-legend": hasSizeLegend,
         [`position-${position}`]: !isBottomPosition,
         // this is required in case
         // position is not BOTTOM but isFluidLegend is true
         "position-bottom": isBottomPosition,
     });
+}
+
+export default function GeoChartLegendRenderer(props: IGeoChartLegendRendererProps): JSX.Element {
+    const { categoryItems, geoData, height, locale, position } = props;
+
+    const availableLegends: IAvailableLegends = getAvailableLegends(categoryItems, geoData);
+    const { hasCategoryLegend, hasColorLegend, hasSizeLegend } = availableLegends;
+    const isLegendVisible = hasCategoryLegend || hasColorLegend || hasSizeLegend;
+    if (!isLegendVisible) {
+        return null;
+    }
+
+    return (
+        <IntlWrapper locale={locale}>
```

Comment:
I don't think we need `IntlWrapper` here since we already wrap it in `VisualizationLoadingHOC` or ([here](https://github.com/gooddata/gooddata-react-components/blob/develop/src/components/core/base/VisualizationLoadingHOC.tsx#L447)).

It could be just
```
const intl = useIntl();
const numericSymbols = getNumericSymbols(intl);
...
```
Resource
- https://github.com/formatjs/react-intl/blob/master/docs/API.md#useintl-hook
- https://github.com/gooddata/gooddata-react-components/blob/develop/src/components/core/base/TranslationsProvider.tsx#L14

Reply:
Removed IntlWrapper.
We can't use react hook in this case, there are alot of if-else, the rules of hook was violated